nixos/modules/config/update-users-groups.pl

```diff
@@ -281,6 +281,11 @@ sub parseUser {
 }
 
 updateFile("/etc/shadow", \@shadowNew, 0600);
+{
+    my $uid = getpwnam "root";
+    my $gid = getgrnam "shadow";
+    chown $uid, $gid, "/etc/shadow";
```

Comment:
```suggestion
    chown($uid, $gid, "/etc/shadow") || die $!;
```
To trigger a fatal error if chown() fails somehow.

Reply:
Done, thanks.

![image](https://user-images.githubusercontent.com/28582702/94293357-fbf73d80-ff12-11ea-9a8b-3b81f7786526.png)